app/models/hbx_enrollment.rb

```diff
@@ -1809,12 +1858,35 @@ def notify_of_coverage_start(publish_to_carrier)
     )
   end
 
+  def term_or_expire_enrollment(term_date = nil)
+    if is_shop?
+      enrollment_benefit_application = sponsored_benefit_package.benefit_application
+      if enrollment_benefit_application.terminated? || enrollment_benefit_application.termination_pending?
+        term_or_cancel_enrollment(self, enrollment_benefit_application.end_on, enrollment_benefit_application.termination_reason)
+      elsif enrollment_benefit_application.expired?
+        expire_coverage!
+      end
+    elsif current_year_ivl_coverage? && term_date.present?
+      terminate_coverage!(term_date)
+    elsif  prior_year_ivl_coverage?
+      expire_coverage!
+    end
+  end
+
+  def fetch_reinstatement_date
+    if coverage_expired?
```

Comment:
if the expired_enrollments feature is turned off in resource_registry then should we still check this condition `coverage_expired`

Reply:
I guess that should be fine, adding a check to see if feature is enabled or not or checking if an enrollment is expired does not make much difference I guess. what do you think?

Reply:
I think depending on the condition you are considering the end date so it will affect whenever we are using this method.